components/board/pi/impl/board.go

```diff
@@ -64,7 +64,10 @@ func init() {
 // accessed via pigpio.
 type piPigpio struct {
 	resource.Named
-	resource.AlwaysRebuild
+	// To prevent deadlocks, we must never lock this mutex while instanceMu, defined below, is
+	// locked. It's okay to lock instanceMu while this is locked, though. This invariant prevents
+	// deadlocks if both mutexes are locked by separate goroutines and are each waiting to lock the
+	// other as well.
 	mu              sync.Mutex
```

Comment:
Does this mu only handle interrupts? Maybe changing the name to `interruptMu`

Reply:
or `gpioMu` if it's handling all of gpio, PWM, spi, i2c and spi

Reply:
The mutex is for GPIO, PWM, interrupts, and anything else in the board that needs to be thread safe. I don't think that has changed from before: it's the mutex for the board object. 

...now that I look closer, though, the trouble is that we're not locking the mutex in all the functions that we need to! For example, `GPIOPinNames` doesn't lock the mutex, and if you call that while you're in the middle of reconfiguration, it's gonna be a mess. I'll go through the whole file and add more mutex locking...